### PR TITLE
Homework_8

### PR DESCRIPTION
**Pull request by mewlunar**

### CODE REVIEW

File: lesson_8_hw/task_2.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+				? 'сотрудник'
+				: 1 < residue && residue < 5
+				? 'сотрудника'
+				: 'сотрудников'
+		}
+	}
+}
+
+function getEnterprisesInfo(enterprises) {
+	const copyEnterprises = JSON.parse(JSON.stringify(enterprises))
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:112]: No need to create a copy here

File: lesson_8_hw/task_2.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+console.log(getEnterprisesInfo(enterprises))
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+function getEnterpriseName(id) {
+	const copyEnterprises = JSON.parse(JSON.stringify(enterprises)).find(
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:145]: Same, no need to copy in this task

File: lesson_8_hw/task_2.js [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+console.log(getEnterpriseName(999))
+console.log(getEnterpriseName('Несуществующий отдел'))
+console.log('==============================================')
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+// Пример:
+// addEnterprise("Название нового предприятия")
+function addEnterprise(name) {
+	const copyEnterprises = JSON.parse(JSON.stringify(enterprises))
+	const newId =
+		Math.max(...copyEnterprises.map(enterprise => enterprise.id)) + 1
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:166]: Id should be unique for all enterprises and departments, so you need fond largest among all ids

File: lesson_8_hw/task_2.js [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+const newEnterpriseName = 'Предприятие 4'
+console.log(formattedEnterpriseObject(addEnterprise(newEnterpriseName)))
+console.log('==============================================')
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+// Пример:
+// addDepartment(1, "Название нового отдела")
+function addDepartment(id, name) {
+	let copyEnterprises = JSON.parse(JSON.stringify(enterprises))
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:187]: Same, no need to copy

File: lesson_8_hw/task_2.js [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+	)
+
+	if (!targetEnterprise) {
+		return `Предприятие с ${id} id не найдено`
+	}
+
+	const allDepartmentIds = copyEnterprises.flatMap(enterprise =>
+		enterprise.departments.map(dep => dep.id)
+	)
+	const newDepartmentId = Math.max(...allDepartmentIds) + 1
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:199]: Same, need to search among all ids, not only in departments

File: lesson_8_hw/task_2.js [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+			const result = departmentId(dept.children, id)
+			if (result) {
+				return result
+			}
+		}
+	}
+	return null
+}
+
+function moveEmployees(id_1, id_2) {
```
Comment (anatoly-karpovich) [inline lesson_8_hw/task_2.js:309]: No check that both departments are from one enterprise

**Review by anatoly-karpovich — APPROVED** [review]
- Remove all copies of company, we are creating methods to update existing array
- create getNewId() method that will search for max id in all enterprises and departments, add +1 and return 
- and please, PLEASE, add Prettier extantion to your VSCode and prettify it)

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**